app/models/king.rb

```diff
@@ -1,7 +1,51 @@
 class King < Piece
+  def valid_move?(to_row, to_col)
+    logger.debug "*** KING: #{id} #{type} valid_move?"
+    return false if move_nil?(to_row, to_col)
+    return false if move_out_of_bounds?(to_row, to_col)
+    return false if move_destination_ally?(to_row, to_col)
+    return false unless move_legal?(to_row, to_col)
+    true
+  end
+
   def move_legal?(to_row, to_col)
+    logger.debug "move_legal for #{type}"
     row_diff = (row - to_row).abs
     col_diff = (col - to_col).abs
-    row_diff <= 1 && col_diff <= 1
+    return true if row_diff <= 1 && col_diff <= 1
+
+    legal_castle_move(to_row, to_col)
+  end
+
+  private
+
+  def legal_castle_move(to_row, to_col)
+    return false unless legal_king_move?(to_row, to_col)
+    legal_rook_move?(to_col)
+  end
+
+  def legal_king_move?(to_row, to_col)
+    return false unless moves.empty?
+    return false unless row == to_row
+    return false unless (col - to_col).abs == 2
+    true
+  end
+
+  def legal_rook_move?(to_col)
```

Comment:
Can you move this into the Rook class and name it `can_castle_to?(to_col)`?

It is essentially logic for the rook, so it makes sense to have it there vs inside the King class.  This would change your `legal_castle_move?(to_col)` method to something like:

```ruby
def legal_castle_move?(to_col)
  rook = get_rook(to_col)
  rook && rook.can_castle_to?(to_col) && can_castle_to?(to_col)
end
```